### ISSUE TITLE
chore: software quality improvement

### ISSUE THREAD

**Issue by soulteary** [maildev/maildev#379]
For some reason, the project has not been actively maintained for a while.

Now we need some volunteers to work together to make this project come alive again, If you are interested, please contact us.

tasks:

- [x] Merge code forked by upstream and downstream projects @soulteary 
  - [x] security fixes
  - [x] dependency upgrade
- [x] Update project test cases to support Node LTS version
- [ ] Handling accumulated issues
  - [x] https://github.com/maildev/maildev/issues/382
  - [x] https://github.com/maildev/maildev/issues/374
  - [x] https://github.com/maildev/maildev/issues/373
  - [x] https://github.com/maildev/maildev/issues/370
  - [x] https://github.com/maildev/maildev/issues/362
  - [x] https://github.com/maildev/maildev/issues/360
  - [x] https://github.com/maildev/maildev/issues/359
  - [x] https://github.com/maildev/maildev/issues/352
  - [x] https://github.com/maildev/maildev/issues/351
  - [x] https://github.com/maildev/maildev/issues/345
  - [x] https://github.com/maildev/maildev/issues/339
  - [x] https://github.com/maildev/maildev/issues/332
  - [ ] https://github.com/maildev/maildev/issues/350
  - [x] https://github.com/maildev/maildev/issues/151 https://github.com/maildev/maildev/issues/150
  - [x] https://github.com/maildev/maildev/issues/326
  - [x] https://github.com/maildev/maildev/issues/354
  - [x] https://github.com/maildev/maildev/issues/363
  - [x] https://github.com/maildev/maildev/issues/310
  - [x] https://github.com/maildev/maildev/issues/356
  - [x] https://github.com/maildev/maildev/issues/396
- [ ] accept pr
  - [x] bcc https://github.com/maildev/maildev/issues/233
  - [x] cli improve https://github.com/maildev/maildev/pull/368 
  - [x] website new ui https://github.com/maildev/maildev/pull/347
  - [x] https://github.com/maildev/maildev/pull/305 https://github.com/maildev/maildev/pull/388
  - [x] https://github.com/maildev/maildev/pull/334 https://github.com/maildev/maildev/pull/389
- [ ] deal with accumulated pr
  - [x] https://github.com/maildev/maildev/pull/344
  - [x] https://github.com/maildev/maildev/pull/376
  - [x] https://github.com/maildev/maildev/pull/371
  - [x] https://github.com/maildev/maildev/pull/364
  - [x] https://github.com/maildev/maildev/pull/357
  - [x] https://github.com/maildev/maildev/pull/355
  - [x] https://github.com/maildev/maildev/pull/353
  - [x] https://github.com/maildev/maildev/pull/328
  - [x] https://github.com/maildev/maildev/pull/314
  - [ ] https://github.com/maildev/maildev/pull/321
  - [x] https://github.com/maildev/maildev/pull/311
  - [x] https://github.com/maildev/maildev/pull/323
- [ ] update the documentation
- [ ] release new version
  - [x] GitHub release
  - [x] DockerHub image upgrade https://github.com/maildev/maildev/issues/382
  - [ ] NPM package upgrade
- [ ] Update github action to automatically publish npm and docker hub

**Comment by idears-org** [comment]
It's great to see this project go back to live again
I have a few queries as well as suggestions here

1. [Queries] Dose "Contributor" got the ability to update docker image and npm package to push updates?
2. [Suggestion] Migrate to TypeScript
3. [Suggestion] Improve Helm chart to store mail file to k8s PV
4. [Suggestion] Pagination support on mail list UI
5. [Suggestion] Multiple types of outgoing support like HTTP

In my case, I have a private fork with some changes and the purpose is to make the project a production-ready proxy to audit all the email sent out form system not just a development smtp server.

**Comment by soulteary** [comment]
Hello @idears-org  👋, thank you for your interest in the project. I try to answer your question:

1.  yes, @djfarrelly will update the software version after double-check. Until then, if someone is in a hurry, they can use the new dockerfile in the repository to build. Or at a later date, I can also provide a version for temporary use.
2. TypeScript is a good suggestion, but the unfinished tasks in this issue may need to be completed first.
3. If there are interested people to contribute together, the progress will be faster https://github.com/maildev/helm 
4. This is also a good suggestion, it looks like it only takes a bit of code to improve the user experience, Talking about the interface, let's continue in another issue https://github.com/maildev/maildev/issues/381
5. This is in my personal plan, but how to implement it into this community version needs to be discussed with other maintainers. Maybe we can open a new issue to discuss in detail?


> In my case, I have a private fork with some changes and the purpose is to make the project a production-ready proxy to audit all the email sent out form system not just a development smtp server.


In my personal private fork, I was originally working on improving this feature, but after being suggested by @djfarrelly , I think making the code public can help public good. Maybe we can play together?

**Comment by idears-org** [comment]
> Maybe we can play together?

Sure, I'll keep my eyes on this project and may also contribute on code base.

**Comment by soulteary** [comment]
We look forward to discussing specific functions and reviewing code with each other in other issues to make the software more practical. 🍻 @idears-org
